snuba/processor.py

```diff
@@ -53,13 +55,18 @@ def _as_dict_safe(value):
     return rv
 
 
-def _collapse_uint32(n):
-    if (n is None) or (n < 0) or (n > MAX_UINT32):
+def _collapse_uint32(n) -> Optional[int]:
+    if n is None:
         return None
-    return n
+
+    i = int(n)
```

Comment:
This ensures this function is actually returning an integer now (not just a float that matches the range.)